### ISSUE TITLE
Full support for callback annotations

### ISSUE THREAD

**Issue by dorranh** [baking-bad/pytezos#262]
Currently when executing `ContractCall.callback_view`, not all of the returned values include their annotations which are specified in the callback's params (for example, see the keys in the output [here](https://github.com/baking-bad/pytezos/blob/fd020d6ebc77fe608cb58bf099b14a58993fdb49/docs/source/quick_start.rst#view-method)). This causes outputs such as in the linked example where a field which should have a name annotation gets assigned a generic name such as `nat_2`. 

It would be great if pytezos could preserve the annotations present in the callback's params. I did a bit of digging into this and unfortunately couldn't quite identify the cause of this issue. Do y'all have any ideas as to what might be causing this behavior?

**Comment by m-kus** [comment]
`callback_view` and `storage_view` methods use builtin REPL which on the one hand allows to skip type checking of the callback contract (and therefore it's not required to deploy a callback contract to the public network), but on the other hand we do not know the exact parameter type of the callback contract (that should include annotations) and therefore cannot properly convert Micheline value to a Python object.

In the example with `balanceOf` the interpreter treats `owner` and `token_id` field annotations as part of the parameter type and preserves them till the end, but it cannot know that the third argument of `TRANSFER_TOKENS` instruction should have `value` annotation.

A workaround for this might be to manually provide the annotated type, but I'm not sure that would be more convenient :)

**Comment by dorranh** [comment]
Thanks for the additional context, @m-kus. That makes sense.

I'm not sure if these annotations are guaranteed to exist, but in the case of some contracts (e.g. the ones created with `ligo`), the contract callback itself does seem to have annotations in the contract's parameters section. I wonder if it would be possible to grab these annotations, if they exist, when calling `callback_view`? For example:

```
{ parameter
    (or (or (or (pair %balance_of
                   (list %requests (pair (address %owner) (nat %token_id)))
                  # This callback has name annotations associated with it
                   (contract %callback
                      (list (pair (pair %request (address %owner) (nat %token_id)) (nat %balance)))))
               ...
```

**Comment by m-kus** [comment]
Hey @dorranh ! This works indeed, thanks for the idea ;)
Will be included in 3.3.1

**Comment by dorranh** [comment]
Awesome, thanks! I'm looking forward to trying it out.
